test/Main2.hs

```diff
@@ -0,0 +1,931 @@
+import StrictEq
```

Comment:
I'm sorry for only now getting back to this pull request.

Everything looks good.. but was there no way of testing the new functionality other than duplicating all of Main.hs to get test open2?  Would there be some way to share the tests between the two opens, or maybe just add a few sanity tests for open2 if sharing is complicated?  Now in order to test direct-sqlite, I'd have to run two test suites instead of one.

Anyway, since it's taken me so long to review this, I can understand if you're not eager to get back to this.  In that case, I might split the patch to include only the sqlite changes and refactor the tests myself.